code/modules/antagonists/blob/blob_mobs.dm

```diff
@@ -145,8 +145,9 @@
 			if(!is_weak && H.stat == DEAD)
 				Zombify(H)
 				break
-	if(!is_valid_z_level(get_turf(src), get_turf(factory)))
-		death()
+	if(!independent)
+		if(!factory || !is_valid_z_level(get_turf(src), get_turf(factory)))
+			death()
```

Comment:
```suggestion
	if(factory && !is_valid_z_level(get_turf(src), get_turf(factory)))
		death()
```
This is what I mean

Reply:
I believe this fits the bill. Also in my testing destroying the factory DID kill the spores, but maybe there's some situation in which it doesn't.